backend/generalhw.pm

```diff
@@ -45,20 +45,10 @@ sub run_cmd ($self, $cmd, @extra_args) {
 
     push @full_cmd, @extra_args;
 
-    my ($stdin, $stdout, $stderr, $ret);
-
-    {
-        # Do not let the SIGCHLD handler of Mojo::IOLoop::ReadWriteProcess::Session steal the exit code from IPC::Run
-        local $SIG{CHLD};
-        eval { $ret = IPC::Run::run(\@full_cmd, \$stdin, \$stdout, \$stderr) };
-        die "Unable to run command '@full_cmd' (deduced from test variable $cmd): $@\n" if $@;
-    }
-    chomp $stdout;
-    chomp $stderr;
-
-    die "$cmd: stdout: $stdout, stderr: $stderr" unless $ret;
-    bmwqemu::diag("$cmd: stdout: $stdout, stderr: $stderr");
-    return $stdout;
+    bmwqemu::diag("Calling $cmd");
+    my $ret = system(@full_cmd);
+    bmwqemu::diag("Command $cmd returned " . ($ret >> 8));
+    die "Failed to run command '@full_cmd' (deduced from test variable $cmd).\n" if ($ret != 0);
```

Comment:
It would be nice to print the error message from `$?` (and avoid the problem mentioned in my other comment):
```suggestion
system(@args) == 0
  or die "Failed to run command '@full_cmd' (deduced from test variable $cmd): $?\n";
```

Reply:
> to print the error message from `$?`

`$?` doesn't contain an error message, only the exit code.

Reply:
I switched to the return code var directly.

Reply:
Which leads to the same behavior as my suggestion (but as I said it should be ok).